pkg/cri/resource-manager/policy/builtin/memtier/pod-preferences.go

```diff
@@ -0,0 +1,290 @@
+// Copyright 2019 Intel Corporation. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package memtier
+
+import (
+	"encoding/json"
+	"strconv"
+	"strings"
+
+	"github.com/ghodss/yaml"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/intel/cri-resource-manager/pkg/cri/resource-manager/cache"
+)
+
+const (
+	// annotation key for opting in to multiple isolated exclusive CPUs per container.
+	keyIsolationPreference = "prefer-isolated-cpus"
+	// annotation key for opting out of exclusive allocation and relaxed topology fitting.
+	keySharedCPUPreference = "prefer-shared-cpus"
+	// annotation key for type of memory to allocate
+	keyMemoryTypePreference = "memory-type"
+)
+
+// types by memory type name
+var memoryNamedTypes = map[string]memoryType{
+	"dram":  memoryDRAM,
+	"pmem":  memoryPMEM,
+	"hbmem": memoryHBMEM,
+	"mixed": memoryAll,
+}
+
+// names by memory type
+var memoryTypeNames = map[memoryType]string{
+	memoryDRAM:  "DRAM",
+	memoryPMEM:  "PMEM",
+	memoryHBMEM: "HBMEM",
+}
+
+// memoryType is bitmask of types of memory to allocate
+type memoryType int
+
+// memoryType bits
+const (
+	memoryUnspec memoryType = 0x0
+	memoryDRAM   memoryType = 0x1
+	memoryPMEM   memoryType = 0x2
+	memoryHBMEM  memoryType = 0x4
+	memoryAll    memoryType = 0x7
```

Comment:
not: https://golang.org/ref/spec#Iota, use `1 << iota` style bitmasks definitions.